### PR TITLE
Topic/concrete hs

### PR DESCRIPTION
**Pull request by bryce-anderson**
I think this is good. The only cost is the translation of ````Seq[(String,String)]```` to whatever type of collection the DSL wants. The concrete type of the collection used by default in this package is an ```ArrayBuffer[(String,String)]``` so it should be reasonably GC friendly.

If nobody has any comments in a few days, I'll just merge this.

### CODE REVIEW

File: http/src/main/scala/org/http4s/blaze/http/http20/Http2StageTools.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  val TE = "te"
+  val ContentLength = "content-length"
+
+  def isLowerCase(str: String): Boolean = {
+    @tailrec
+    def go(i: Int): Boolean = {
+      if (i == str.length) true
+      else {
+        val ch = str.charAt(i)
+        if ('A' <= ch && ch <= 'Z') false
```
Comment (rossabaker) [inline http/src/main/scala/org/http4s/blaze/http/http20/Http2StageTools.scala]: What about diacritics, or are those not valid here?

**Comment by rossabaker** [comment]
I'm not deeply up to speed on the architecture here, but it seems reasonable.

File: http/src/main/scala/org/http4s/blaze/http/http20/Http2StageTools.scala (hunk 1 above)
Comment (bryce-anderson) [inline http/src/main/scala/org/http4s/blaze/http/http20/Http2StageTools.scala]: to be very strict I should check to make sure its lower case ascii. This was done just so I could match against the special headers as Strings.

**Comment by bryce-anderson** [comment]
Thanks for the comments.

**Pull request merged**